### PR TITLE
Add .NET 10 SDK to build.yml and fix flaky OOProc smoke tests

### PR DESCRIPTION
**Pull request by bachuv**
# Summary
## What changed?

This pull request updates the build pipeline to support .NET 10 SDK and addresses assembly conflicts in end-to-end test projects by explicitly targeting .NET 8.0 during the build of a shared project. These changes ensure compatibility with newer SDKs while maintaining stable test builds.

Build pipeline updates:

* Added a step in the build pipeline (`build.yml`) to install and use the .NET 10 SDK, enabling builds and tests against the latest .NET version.

Test project build fixes:

* Updated `extensions.csproj` files for end-to-end test apps (`BasicJava`, `BasicNode`, `BasicPowerShell`, `BasicPython`) to pin the build and pack step of `WebJobs.Extensions.DurableTask.csproj` to target `net8.0`. This avoids MSB3277 assembly conflicts when the source project multi-targets both `net8.0` and `net10.0`.

## Why is this change needed?
- Supports .NET 10 in our official pipeline
- Fixes flaky tests

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): VS Code GitHub CoPilot
- AI-assisted areas/files: build.yml and extensions.csproj files
- What you changed after AI output:

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed / Failed (link logs if failed)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components):
- Steps + observed results:
  1.
  2.
  3.
- Evidence (optional):

---

# Notes for reviewers
- N/A

### CODE REVIEW

File: eng/templates/build.yml [hunk 1]
```diff
@@ -36,6 +36,12 @@ jobs:
             packageType: 'sdk'
             version: '8.0.x'
 
+      - task: UseDotNet@2
+        displayName: 'Use the .NET 10 SDK'
+        inputs:
+            packageType: 'sdk'
+            version: '10.0.x'
```
[the remark this answers is not in the document]
Comment (bachuv) [reply eng/templates/build.yml:43]: Keeping this change. The repo's `global.json` already pins to .NET 10:

```json
{
  "sdk": {
    "version": "10.0.102",
    "rollForward": "latestFeature"
  }
}

File: test/e2e/Apps/BasicPython/extensions.csproj [hunk 2]
```diff
@@ -6,10 +6,11 @@
 	<DefaultItemExcludes>**</DefaultItemExcludes>
   </PropertyGroup>
   <Target Name="PreBuild" BeforeTargets="Build;BeforeResolveReferences">
-    <!-- Build and pack the source project -->
+    <!-- Build and pack the source project. Pin to net8.0 to avoid MSB3277 assembly conflicts
+         when the source project multi-targets net8.0;net10.0 -->
     <MSBuild Projects="$(MSBuildProjectDirectory)/../../../../src/WebJobs.Extensions.DurableTask/WebJobs.Extensions.DurableTask.csproj"
              Targets="Build;Pack"
-             Properties="Configuration=$(Configuration);PackageOutputPath=$(MSBuildProjectDirectory)/packages" />
+             Properties="Configuration=$(Configuration);PackageOutputPath=$(MSBuildProjectDirectory)/packages;TargetFramework=net8.0" />
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Apps/BasicPython/extensions.csproj]: Keeping these comments.

File: test/e2e/Apps/BasicPython/extensions.csproj [hunk 3]
```diff
@@ -6,10 +6,11 @@
 	<DefaultItemExcludes>**</DefaultItemExcludes>
   </PropertyGroup>
   <Target Name="PreBuild" BeforeTargets="Build;BeforeResolveReferences">
-    <!-- Build and pack the source project -->
+    <!-- Build and pack the source project. Pin to net8.0 to avoid MSB3277 assembly conflicts
+         when the source project multi-targets net8.0;net10.0 -->
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Apps/BasicPython/extensions.csproj]: Fixed.

**Review by andystaples — APPROVED** [review]

**Review by andystaples — CHANGES_REQUESTED** [review]
Hang on - changing to change requested while I look at one thing

**Comment by andystaples** [comment]
Opened a PR to this PR that allows us to respect the chosen TFM for non-dotnet apps too
https://github.com/Azure/azure-functions-durable-extension/pull/3364/changes

**Review by torosent — APPROVED** [review]

File: test/e2e/Tests/build-e2e-test.ps1 [hunk 4]
```diff
@@ -132,12 +132,22 @@ function InstallExtensionAndBuildTestApp($testAppDir) {
...
-      else {
-        Write-Warning "func command not found. Skipping extensions sync."
+      if (Test-Path ".\extensions.csproj") {
+        Write-Host "Building extensions project"
+        dotnet clean extensions.csproj
+        if ($TargetFramework) {
+          dotnet build extensions.csproj -p:TargetFramework=$TargetFramework
+        } else {
+          dotnet build extensions.csproj
+        }
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Tests/build-e2e-test.ps1]: Fixed.

File: test/e2e/Tests/build-e2e-test.ps1 [hunk 5]
```diff
@@ -132,12 +132,22 @@ function InstallExtensionAndBuildTestApp($testAppDir) {
...
-      }
-      else {
-        Write-Warning "func command not found. Skipping extensions sync."
+      if (Test-Path ".\extensions.csproj") {
+        Write-Host "Building extensions project"
+        dotnet clean extensions.csproj
+        if ($TargetFramework) {
+          dotnet build extensions.csproj -p:TargetFramework=$TargetFramework
+        } else {
+          dotnet build extensions.csproj
```
[the remark this answers is not in the document]
Comment (bachuv) [reply test/e2e/Tests/build-e2e-test.ps1]: Fixed.

**Review by andystaples — APPROVED** [review]
